File: solutions/cpp/resistor-color/1/resistor_color.cpp
#include "resistor_color.h"
#include <cctype>
#include <stdexcept>
#include <algorithm>
namespace resistor_color {

    // TODO: add your solution here
    const std::vector<std::string> ColorNames{
        "black",
        "brown",
        "red",
        "orange",
        "yellow",
        "green",
        "blue",
        "violet",
        "grey",
        "white",
    };
    int color_code(std::string color_name) {
        std::transform(color_name.begin(), color_name.end(), color_name.begin(), std::tolower);
        if (auto it = std::find(ColorNames.cbegin(), ColorNames.cend(), color_name); it != ColorNames.cend()) {
            return static_cast<int>(std::distance(ColorNames.cbegin(), it));
        } else {
            throw std::domain_error("invalid color name.");
        }
    }
    std::vector<std::string> colors() {
        return ColorNames;
    }

}  // namespace resistor_color

File: solutions/python/resistor-color/1/resistor_color.py
RESISTER_COLOR: dict[str, int] = {
    "black": 0,
    "brown": 1,
    "red": 2,
    "orange": 3,
    "yellow": 4,
    "green": 5,
    "blue": 6,
    "violet": 7,
    "grey": 8,
    "white": 9,
}

def color_code(color: str) -> int:
    return RESISTER_COLOR[color]

def colors() -> list[str]:
    return list(RESISTER_COLOR)
